File: 0048-rotate-image/0048-rotate-image.java
class Solution {
    public void rotate(int[][] matrix) {
        for(int i=0;i<matrix.length;i++){
            for(int j=i+1;j<matrix.length;j++){
                int temp=matrix[i][j];
                matrix[i][j]=matrix[j][i];
                matrix[j][i]=temp;
            }
        }
        for(int i = 0; i < matrix.length; i++) {

            int j = matrix[i].length - 1;
            int k = 0;

            while(k < j) {

                int temp = matrix[i][k];
                matrix[i][k] = matrix[i][j];
                matrix[i][j] = temp;

                k++;
                j--;
            }
        }
    }
}

File: 0048-rotate-image/0048-rotate-image.py
class Solution(object):
    def rotate(self, matrix):
        for i in range(len(matrix)):
            for j in range(i+1,len(matrix)):
                matrix[i][j],matrix[j][i]=matrix[j][i],matrix[i][j]
        for i in range(len(matrix)):
            j=len(matrix[i])-1
            k=0
            while(k<j):
                matrix[i][k],matrix[i][j]=matrix[i][j],matrix[i][k]
                k=k+1
                j=j-1
